Fix dbcopy client check_host & handle exceptions

luca-drf opened the pull request.

thibauthourlier commented:
It looks like Thomas fixed the bug already... https://github.com/Ensembl/ensembl-prodinf-core/commit/891fcbbc418b61e72c84a8e9374ad67bb3bc3dd0

marcoooo commented:
> It looks like Thomas fixed the bug already... [891fcbb](https://github.com/Ensembl/ensembl-prodinf-core/commit/891fcbbc418b61e72c84a8e9374ad67bb3bc3dd0)

SO I close this PR then.
Thanks

Commit:
Fix missing logging import in dbcopy.py

Commit:
Override rest retrieve_job and list_jobs in dbcopy

This way code using DbCopyRestClient won't need to import requests to
handle requests exceptions

Commit:
Fix _check_host domain logic in dbcopy

marcoooo commented on src/ensembl/production/core/clients/dbcopy.py:
```diff
@@ -117,7 +143,7 @@ def _check_host(self, url, host_port_map):
         host_parts = host.split('.')
         if len(host_parts) > 1:
-            if not host.endswith('.ebi.ac.uk'):
+            if host.endswith('.ebi.ac.uk'):
```
Not sure about this update. I see your point when calling `.ensembl.org`,  we want to insure that the call is made within ebi. But this change makes the script not usable on `ensprod-dev-01.ebi.ac.uk` if accepted.

luca-drf replied:
@marcoooo This check is to avoid users specifying a fully qualified hostname for source or target db server, which the DBAs script would silently ignore.
